component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -281,6 +285,25 @@ private void addOrValidateCertificate(String userName, AuthenticationContext aut
             throw new AuthenticationFailedException("Error in validating the user certificate", e);
         }
 
+        try {
+            // Check whether user account is locked or not.
+            if (isAccountLock(userName)) {
+                authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                        X509CertificateConstants.USER_ACCOUNT_LOCKED);
+                throw new AuthenticationFailedException("Account is locked for user: " + userName);
+            }
+
+            // Check whether user account is disable or not.
+            if (isAccountDisabled(getUsername(authenticationContext))) {
+                authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                        X509CertificateConstants.USER_ACCOUNT_DISABLED);
+                throw new AuthenticationFailedException("Account is disabled for user: " + userName);
+            }
+        } catch (UserStoreException | AccountLockServiceException e) {
```

Comment:
Don't we need to throw the `e` with the exception?

Reply:
Fixed with ed2968b2c84d173790ff71b500a0268434981d01